addons/binding/org.openhab.binding.sinope/src/main/java/org/openhab/binding/sinope/handler/SinopeThermostatHandler.java

```diff
@@ -0,0 +1,279 @@
+                    e.getLocalizedMessage());
+            gateway.setCommunicationError(true);
+
+        }
+        gateway.schedulePoll();
+    }
+
+    public void setSetpointTemp(SinopeThermostatHandler sinopeThermostatHandler, float temp)
+            throws UnknownHostException, IOException {
+        int newTemp = (int) (temp * 100.0);
+        byte[] deviceId = SinopeConfig.convert(sinopeThermostatHandler.getDeviceId());
+        if (deviceId == null) {
+            logger.error("Invalid Device id: {}", deviceId);
+            updateStatus(ThingStatus.UNINITIALIZED);
+            return;
+        }
+
+        SinopeGatewayHandler gateway = getSinopeGatewayHandler();
+        if (gateway.connectToBridge()) {
+
+            logger.debug("Connected to bridge");
+
+            SinopeDataWriteRequest req = new SinopeDataWriteRequest(gateway.newSeq(), deviceId,
+                    new SinopeSetPointTempData());
+            ((SinopeSetPointTempData) req.getAppData()).setSetPointTemp(newTemp);
+
+            SinopeDataAnswer answ = (SinopeDataAnswer) gateway.execute(req);
+
+            if (answ.getStatus() == 0) {
+                logger.debug("Setpoint temp is now: {} C", newTemp);
+            } else {
+                logger.debug("Cannot Setpoint temp, status: {}", answ.getStatus());
+            }
+        } else {
+            logger.error("Could not connect to bridge to update Setpoint Temp");
+        }
+
+    }
+
+    public void setSetpointMode(SinopeThermostatHandler sinopeThermostatHandler, int mode)
+            throws UnknownHostException, IOException {
+        byte[] deviceId = SinopeConfig.convert(sinopeThermostatHandler.getDeviceId());
+        if (deviceId == null) {
+            logger.error("Invalid Device id: {}", deviceId);
+            updateStatus(ThingStatus.UNINITIALIZED);
+            return;
+        }
+
+        SinopeGatewayHandler gateway = getSinopeGatewayHandler();
+        gateway.stopPoll();
+
+        if (gateway.connectToBridge()) {
+            logger.debug("Connected to bridge");
+
+            SinopeDataWriteRequest req = new SinopeDataWriteRequest(gateway.newSeq(), deviceId,
+                    new SinopeSetPointModeData());
+            ((SinopeSetPointModeData) req.getAppData()).setSetPointMode((byte) mode);
+
+            SinopeDataAnswer answ = (SinopeDataAnswer) gateway.execute(req);
+
+            if (answ.getStatus() == 0) {
+                logger.debug("Setpoint mode is now : {}", mode);
+            } else {
+                logger.debug("Cannot Setpoint mode, status: {}", answ.getStatus());
+            }
+        } else {
+            logger.error("Could not connect to bridge to update Setpoint Mode");
+        }
+        gateway.schedulePoll();
+
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("bridgeStatusChanged {}", bridgeStatusInfo);
+        updateStatus(bridgeStatusInfo.getStatus());
```

Comment:
If the bridge goes online the thing also goes online, even if it has a wrong device id for example?

Reply:
Valid concern of @triller-telekom. @chaton78 Did you consider this?